test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs

```diff
@@ -443,4 +456,33 @@ public void ShouldIgnoreStaticConstructorAndDiagnostic()
             .HaveDiagnostic(DiagnosticDescriptors.CouldNotCreateMapping)
             .HaveAssertedAllDiagnostics();
     }
+
+    [Fact]
+    public void ShouldFlattedNestedToTargetRootMapping()
```

Comment:
Could you move these tests to `ObjectPropertyFlatteningTest` as they fit better there as this always flattens the values.